### PR TITLE
Feature/dcc 4749 repository index support

### PR DESCRIPTION
**Pull request by btiernay**

### CODE REVIEW

File: dcc-portal-ui/app/scripts/repository/js/services.js [hunk 1]
```diff
@@ -77,7 +67,12 @@
       _repoMapRefreshPromise = _getRepoMap();
 
       _repoMapRefreshPromise.then(function (restangularMapData) {
-        var repoMapData = restangularMapData.plain();
+
```
Comment (btiernay) [inline dcc-portal-ui/app/scripts/repository/js/services.js]: No need!

Comment (btiernay) [inline dcc-portal-ui/app/scripts/repository/js/services.js]: Called it!

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/resource/entity/FileResource.java [hunk 2]
```diff
@@ -102,14 +102,6 @@ public Response legacyManifest(
     return Response.seeOther(uri).build();
   }
 
-  @GET
-  // TODO: This should just be /repositories. ps don't use underscores!!
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/resource/entity/FileResource.java]: Like this comment!

@ithinkicancode Tisk tisk!

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/repository/RepositoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import lombok.SneakyThrows;
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+public class RepositoryRepository {
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/repository/RepositoryRepository.java]: Wow!

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/model/RepositoryServer.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import static lombok.AccessLevel.PRIVATE;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor(access = PRIVATE)
+public enum RepositoryServer {
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/model/RepositoryServer.java]: This should probably go away now and we should just use `Repository` to make this more dynamic and reduce duplication. Will do in another iteration.

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/repository/RepositoryRepository.java (hunk 3 above)
Comment (andricDu) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/repository/RepositoryRepository.java]: Maybe refactor it into a RepositoryRepositoryDaoFactory? 😜

**Pull request merged**